packages/storybook/src/generators/configuration/configuration.ts

```diff
@@ -107,7 +107,7 @@ export async function configurationGenerator(
     projectIsRootProjectInStandaloneWorkspace(root),
     !!nextBuildTarget,
     compiler === 'swc',
-    !!viteBuildTarget,
+    !!viteBuildTarget || schema.uiFramework.endsWith('-vite'),
```

Comment:
I've approved the PR, but it might be more valid in the future to have a Set containing the known `-vite` frameworks.  This can allow us to determine stability and support for new frameworks, however, it does prevent people using a third party package that we have not specifically added support for. 

It comes down to whether we want to ensure stability or let users break things